javascripts/discourse/initializers/init-alt-vote-cat-style.js

```diff
@@ -46,114 +46,120 @@ export default apiInitializer("1.1", (api) => {
-
-          if (target.classList.contains("can-vote")) {
-            voteCountElem.innerHTML = voteCount + 1;
-            voteType = "vote";
-
-            this.set("votesLeft", this.votesLeft - 1);
-            this.set("userVoted", true);
-
-            if (this.votesLeft === 0) {
-              document
-                .querySelectorAll(
-                  ".topic-list-item:not(.closed) .topic-list-vote-button.can-vote"
-                )
-                .forEach((voteButton) => {
-                  const tli = voteButton.closest(".topic-list-item");
-                  if (
-                    tli.classList.contains("visited") ||
-                    !tli.classList.contains("unseen-topic")
-                  ) {
-                    voteButton.classList.add("disabled");
-                    voteButton.title = i18n(themePrefix("out_of_votes"));
-                  }
-                });
-            }
-            // Ensure clicked button has proper title and class
-            target.title = i18n(themePrefix("user_vote"));
-            target.classList.remove("disabled");
-          } else {
-            voteCountElem.innerHTML = voteCount - 1;
-            voteType = "unvote";
-
-            this.set("votesLeft", this.votesLeft + 1);
-            this.set("userVoted", false);
-
-            if (this.votesLeft > 0) {
-              document
-                .querySelectorAll(
-                  ".topic-list-item:not(.closed) .topic-list-vote-button.can-vote"
-                )
-                .forEach((voteButton) => {
-                  const tli = voteButton.closest(".topic-list-item");
-                  if (
-                    tli.classList.contains("visited") ||
-                    !tli.classList.contains("unseen-topic")
-                  ) {
-                    voteButton.classList.remove("disabled");
-                    voteButton.title = i18n(themePrefix("user_no_vote"));
-                  }
-                });
+    api.modifyClass(
+      "component:topic-list-item",
+      (Superclass) =>
+        class extends Superclass {
+          @service("router") excerptsRouter;
+          @alias("currentUser.votes_left") votesLeft;
+          @alias("topic.user_voted") userVoted;
+
+          @discourseComputed("topic")
+          unboundClassNames(topic) {
+            let classList = super.unboundClassNames;
+            if (!topic.can_vote) {
+              classList += " non-voting";
             }
+            return classList;
+          }
 
-            target.title = i18n(themePrefix("user_no_vote"));
+          @discourseComputed(
+            "excerptsRouter.currentRoute.attributes.category.id"
+          )
+          expandPinned(currentCategoryId) {
+            return currentCategoryId &&
+              (settings.include_excerpts || settings.vote_from_topic_list) &&
+              votingCategories.some((c) => c === Number(currentCategoryId))
+              ? true
+              : super.expandPinned();
```

Comment:
should this be

```suggestion
              : super.expandPinned;
```

like `super.unboundClassNames` above?